GTViewController/Classes/Nav Classes/GTFollowupViewController.m

```diff
@@ -296,4 +270,55 @@ - (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)butto
     [alertView dismissWithClickedButtonIndex:buttonIndex animated:YES];
 }
 
+
+#pragma mark - UISnuffleButtonTapDelegate methods
+
+- (void)didReceiveTapOnPositiveButton:(UISnuffleButton *)positiveButton {    
+    NSArray *inputValidationErrors = [self inputValidationErrors];
+    
+    if ([positiveButton forceValidation] && [inputValidationErrors count]) {
+        [[[UIAlertView alloc] initWithTitle:@"Please correct these fields:"
+                                    message:[inputValidationErrors componentsJoinedByString:@"\n"]
+                                   delegate:self
+                          cancelButtonTitle:@"Ok"
+                          otherButtonTitles:nil] show];
+        return;
+    }
+    
+    [positiveButton.tapEvents enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
It makes the renderer more general and capable of more layouts if the event logic stays in the button class. See my comment above for an example.